### PR TITLE
Align V1 receipt endpoint with V2/V3 reveal rules for secrets

### PR DESCRIPTION
**Pull request by delano**
## Summary

The V1 API receipt endpoint now aligns with V2/V3 behavior by restricting secret revelation on receipts to generated values only, and only on first view within a configured display window. Concealed (user-supplied) plaintext is never revealed on the receipt, closing a long-standing inconsistency where the receipt endpoint could be used to sidestep the at-most-once rule.

## Changes

**Logic (`show_receipt.rb`)**
- Modified secret value revelation to check:
  - Receipt is in `:new` state (first view)
  - Secret is generated (not concealed/user-supplied)
  - Receipt age is within `site.secret_options.generated_value_display_ttl`
- Concealed secrets no longer have their plaintext decrypted for the receipt response
- Added logging for generated secret decryption

**Tests (`show_receipt_spec.rb`)**
- Updated test fixtures to support `kind` parameter (generate vs. conceal)
- Added context for concealed secrets: verifies plaintext is not revealed despite being decryptable
- Added context for generated secrets with three scenarios:
  - First view within display window: value is revealed
  - After receipt is previewed: value is not revealed
  - Outside display window: value is not revealed
- Updated mock receipt to include `kind` and `created` attributes

**Changelog**
- Documented the behavior change and rationale

## Test Plan

The change is covered by comprehensive unit tests:
- Existing tests verify no errors are raised during processing
- New tests verify concealed secrets are never decrypted for receipts
- New tests verify generated secrets are only revealed on first view within the display window
- All test scenarios pass with the updated logic

https://claude.ai/code/session_01X22mUEhk4soWdtWW8nTugb

### CODE REVIEW

File: apps/api/v1/spec/logic/secrets/show_receipt_spec.rb [hunk 1]
```diff
@@ -52,20 +56,54 @@
...
+
+        subject.process
+
+        expect(subject.secret_value).to be_nil
+      end
+
+      it 'does not reveal the value outside the display window' do
+        display_ttl = OT.conf.dig('site', 'secret_options', 'generated_value_display_ttl').to_i
+        receipt.created = Familia.now.to_i - (display_ttl + 10)
+        receipt.save
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/api/v1/spec/logic/secrets/show_receipt_spec.rb:102]: Applied in `1515042`. Added the explicit `expect(display_ttl).to be_positive` guard so the "outside display window" test fails loudly if the config TTL is ever `0` in CI, rather than silently exercising the `display_ttl.positive?` short-circuit branch.

Also added a companion unit test (`does not decrypt when generated_value_display_ttl is zero or unset`) that stubs the config to return `0` and asserts `decrypted_secret_value` is never called — so the zero-TTL branch is now covered intentionally rather than by accident.

---
_Generated by [Claude Code](https://claude.ai/code)_
